sdk/core/azure-core/tests/test_authentication.py

```diff
@@ -638,3 +660,47 @@ def verify_authorization_header(request):
     pipeline = Pipeline(transport=transport, policies=[credential_policy])
 
     pipeline.run(http_request("GET", "https://test_key_credential"))
+
+
+def test_access_token_unpack():
+    """Test various unpacking of AccessToken."""
+    token = AccessToken("token", 42)
+    assert token.token == "token"
+    assert token.expires_on == 42
+    assert token.refresh_on is None
+
+    token, expires_on = AccessToken("token", 42)
+    assert token == "token"
+    assert expires_on == 42
+
+    token, expires_on = AccessToken(token="token", expires_on=42)
+    assert token == "token"
+    assert expires_on == 42
+
+    token, expires_on, refresh_on = AccessToken("token", 42, 21)
+    assert token == "token"
+    assert expires_on == 42
+    assert refresh_on == 21
+
+    token, expires_on, refresh_on = AccessToken(token="token", expires_on=42, refresh_on=21)
+    assert token == "token"
+    assert expires_on == 42
+    assert refresh_on == 21
+
```

Comment:
We also want to test the case:

token, expires_on = AccessToken(token="token", expires_on=42, refresh_on=21)
assert token == "token"
assert expires_on == 42